sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/_tests_/e2e/application-offering-change-request.institutions.controller.getCompletedApplications.e2e-spec.ts

```diff
@@ -0,0 +1,260 @@
+    );
+  });
+
+  it(`Should return all completed application offering change requests for the location in ${FieldSortOrder.ASC} order of name when requested.`, async () => {
+    // Arrange
+
+    // Student 1 has a completed application to the institution.
+    await saveFakeApplicationDisbursements(
+      db.dataSource,
+      { institutionLocation: collegeFLocation },
+      {
+        applicationStatus: ApplicationStatus.Completed,
+      },
+    );
+    // Student 2 has an in-progress with student application offering a change request for the institution.
+    await saveFakeApplicationOfferingRequestChange(db.dataSource, {
+      institutionLocation: collegeFLocation,
+    });
+    // Student 3 has an approved application offering change request for the institution.
+    const applicationOfferingChange3 =
+      await saveFakeApplicationOfferingRequestChange(
+        db.dataSource,
+        {
+          institutionLocation: collegeFLocation,
+        },
+        {
+          applicationOfferingChangeRequestStatus:
+            ApplicationOfferingChangeRequestStatus.Approved,
+        },
+      );
+    // Student 4 has an in progress with student application offering change request with a different institution.
+    await saveFakeApplicationOfferingRequestChange(db.dataSource);
+    // Student 5 has a declined by student application for the institution, that have an approved application offering change request.
+    const applicationOfferingChange5 =
+      await saveFakeApplicationOfferingRequestChange(
+        db.dataSource,
+        {
+          institutionLocation: collegeFLocation,
+        },
+        {
+          applicationOfferingChangeRequestStatus:
+            ApplicationOfferingChangeRequestStatus.DeclinedByStudent,
+        },
+      );
+    // Student 6 has a declined by student application for the institution, that have an approved application offering change request.
+    const applicationOfferingChange6 =
+      await saveFakeApplicationOfferingRequestChange(
+        db.dataSource,
+        {
+          institutionLocation: collegeFLocation,
+        },
+        {
+          applicationOfferingChangeRequestStatus:
+            ApplicationOfferingChangeRequestStatus.DeclinedBySABC,
+        },
+      );
+
+    applicationOfferingChange3.assessedDate = new Date();
+    applicationOfferingChange5.studentActionDate = new Date();
+    applicationOfferingChange6.assessedDate = new Date();
+
+    await db.applicationOfferingChangeRequest.save([
+      applicationOfferingChange3,
+      applicationOfferingChange5,
+      applicationOfferingChange6,
+    ]);
+
+    const application3 = applicationOfferingChange3.application;
+    const application5 = applicationOfferingChange5.application;
+    const application6 = applicationOfferingChange6.application;
+
+    application3.applicationNumber = "1000000000";
+    application5.applicationNumber = "1000000001";
+    application6.applicationNumber = "1000000002";
```

Comment:
The application numbers used across `getCompletedApplications` and `getEligibleApplications` are the same. In future, if a get request (find) request is made using the application numbers, it can cause the test to fail due to interference with other tests. Can we use random application numbers to avoid this issue?

Reply:
I believe we have been using this across the tests. won't the db be reset for tests? and even random will not be making sure that every application number is unique right?

Reply:
The application numbers are usually created and accessed at the "student scope". A future `find` would usually include the student also which would avoid the conflicts.

Reply:
Sounds good 👍

Reply:
@ann-aot Random (from a large pool say million) in my understanding will have a negligible likelihood of collision. I don't think that the data is reset for every test suite. Either way, since @andrewsignori-aot mentioned that it will be student scoped, it should not result in a conflict.

Reply:
@sh16011993  Here, I did a small probability test, 10000 with 100 times, so the chance probability is 0.01 here (that gave me a duplicate value on my 2nd run), 
![image](https://github.com/bcgov/SIMS/assets/77353155/b563de24-233d-4f86-8e5e-13183a811682)
so, that means, there is a possibility for generating the same value multiple times, It may happen at 2nd or 100th or 1000th  time, but there is a possibility. and I believe that this project is a long-run project, which may at least have a maintenance pull request periodically, so, as mentioned, random does not ensure that the value is unique. But I believe UUID and .unquiue may ensure, not sure.  But I don't think we can use UUID in this case. 

` I don't think that the data is reset for every test suite.`: I was not saying about each test suite, I believe that e2e tests when on each commit in Git. there is a db reset happing. I may be wrong.

But as the dev mentioned, for a scenario like which u have mentioned, we can either find it with application number +id or any other condition according to the requirement